src/daft-core/src/datatypes/field.rs

```diff
@@ -129,3 +129,16 @@ impl Display for Field {
         write!(f, "{}#{}", self.name, self.dtype)
     }
 }
+
+impl PartialEq for Field {
```

Comment:
I implemented a custom `PartialEq` and `Hash` for `Field` because we were getting a bunch of issues with `Schema::eq` in both Rust and Python, now that our arrow2 reader is propagating `field_id` into the metadata field.

Not sure if this is the best idea though. Reviewers should feel free to comment!